docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
I'm slightly negative on having a "key" member variable and forcing the caller to know that it is a private key and they need to call a public_key() method on it to convert it to something they can exchange with their peer. It seems like the key exchange class should be providing the value to be exchanged directly. I'd be ok with naming the method "public_key()" instead of "get_public()", having the call be on the ecdh object and not a member of it, but my general leaning is to make method names be actions rather than nouns.

On the other point, I'm not a big fan of the agree() method name, but I agree with the point about making the name more clearly indicate a computation is occurring. Perhaps this could be something like compute_shared() or construct_shared(). Since the KDF is being passed as an argument there, passing the wrong KDF would be no different than having the wrong private or public key. It'll only be the same shared key if all three of those things are consistent on the two sides, but no matter what SOME key that could be shared will be returned.